addons/website_sale/tests/test_website_sale_cart.py

```diff
@@ -63,6 +66,40 @@ def test_add_cart_archived_product(self):
             with MockRequest(product.with_user(self.public_user).env, website=self.website.with_user(self.public_user)):
                 self.WebsiteSaleController.cart_update_json(product_id=product.id, add_qty=1)
 
+    def test_zero_price_product_rule(self):
```

Comment:
Wdym more useful? It'd be redundant to test the same function in each separate module ; rather, it's better to test in the same module where the logic is defined and ensure the function works with any kind of usage

Reply:
From a technical POV, it's nice to test the architecture/api, like it's done here.
But what we fix in this PR is not the behavior of the api, but specific business flows.

This test would still work even if I remove your overrides (even if it would be dumb, I agree, but let's say it's done by mistake), so we could potentially break again what you're fixing without making your tests red (unless we remove the base method itself).

Anyway, I won't block this PR for that, I wouldn't say that this test is wrong or bad, but it's not enough if you want to cover your overrides.